config/routes.rb

```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
   resources :categories, only: [:show]
   resources :posts, only: [:show]
+  resources :users, only: [:create, :new]
+
+  get "login", to: "sessions#new"
+  post "login", to: "sessions#create"
+  get "logout", to: "sessions#destroy"
```

Comment:
What do you think about using a resource route for the session instead?

```ruby
resources :session, only: [:new, :create, :destroy]
```

Reply:
I wanted the URL to end with `/login` etc, which is why I went with this option.  Is there a way to uses a resource route and override the naming convention?

Reply:
Ah no, in that case your custom routes are the way to go. I would suggest making the logout route a `POST` or `DELETE` though.